File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        if(s.size()==1){
            return 1;
        }
        set<char> st;
        int j=0;
        int mx=0;
        for(int i=0;i<s.size();i++){
            //cout<<"i-"<<i<<endl;
            mx=max(mx,i-j);
            if(st.count(s[i])){
                while(st.count(s[i])){
                    st.erase(s[j]);
                    j++;
                }
            }
            st.insert(s[i]);
        }
        int q=s.size();
        mx=max(mx,q-j);
        
        return mx;
    }
};

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        res=0
        l=0
        cr = set()
        for r in range(len(s)):
            while s[r] in cr:
                cr.remove(s[l])
                l+=1
            cr.add(s[r])
            res=max(res,r-l+1)
        return res
